lib/Doctrine/ORM/Query/Lexer.php

```diff
@@ -164,7 +164,7 @@ protected function getType(&$value)
                 return self::T_STRING;
 
             // Recognize identifiers
-            case (ctype_alpha($value[0]) || $value[0] === '_'):
+            case (ctype_alpha($value[0]) || $value[0] === '_' || $value[0] === '\\'):
```

Comment:
After looking more carefully at [DDC-3701](http://www.doctrine-project.org/jira/browse/DDC-3701), I don't think that this fix is needed. In DQL, an identifier is basically anything "referenceable" (alias or entity name). An entity name cannot start with `\`, because DQL has no namespacing.